### ISSUE TITLE
[BUG] 

### ISSUE THREAD

**Issue by wkelly1** [francoismassart/eslint-plugin-tailwindcss#287]
**Describe the bug**
Tailwind prefix is ignored when the `tailwindcss/no-custom-classname` rule is disabled.

**To Reproduce**
1. Disable the `tailwindcss/no-custom-classname` eslint rule in `.eslintrc` by adding ```"tailwindcss/no-custom-classname": "off"``` to rules
2. In the tailwind config specify a prefix such as `prefix: "tw-"`
3. Add a class that will cause another tailwindcss eslint error. For example `tw-flex-shrink-0`. This will accuratelly report that the it should be updated to `tw-shrink-0` in Tailwindcss 3.
4. Do the same as 3 but add `flex-shrink-0`. This would normally be caught by the `tailwindcss/no-custom-classname` rule however when disabled it also reports that it should be updated to `shrink-0` via the `tailwindcss/migration-from-tailwind-2` rule.

**Expected behavior**
No eslint warning should be output when the class `flex-shrink-0` is added. Instead that should be ignored and only when `tw-flex-shrink-0` is written should the `tailwindcss/migration-from-tailwind-2` be outputed.

**Screenshots**
![ESLint tailwind error](https://github.com/francoismassart/eslint-plugin-tailwindcss/assets/26680240/77a84d5a-90a4-45f8-b1ce-b6c53758ebfd)

**Environment (please complete the following information):**
 - OS: [macOS, windows 10]
 - Softwares + version used:
     -  VSCode: `1.81.2`
     -  eslint-plugin-tailwindcss: `3.13.0`
     -  eslint: `^8.45.0`

**Additional context**

Tailwindcss config:

```js
/** @type {import('tailwindcss').Config} */
export default {
  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
  theme: {
    extend: {},
  },
  plugins: [],
  prefix: "tw-",
};
```

**eslint config file or live demo**
eslint config file:
```js
module.exports = {
  root: true,
  env: { browser: true, es2020: true },
  extends: [
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:react-hooks/recommended",
    "plugin:tailwindcss/recommended",
  ],
  ignorePatterns: ["dist", ".eslintrc.cjs"],
  parser: "@typescript-eslint/parser",
  plugins: ["react-refresh"],
  rules: {
    "react-refresh/only-export-components": [
      "warn",
      { allowConstantExport: true },
    ],
    "tailwindcss/no-custom-classname": "off",
  },
};
```

**Comment by francoismassart** [comment]
Closing this issue, cleaning up before release of v4.
